### PR TITLE
Added a command that resets to the last progress trigger (Issue #897)

### LINKED ISSUE #897: Add a command that resets to the last progress trigger
**Is your feature request related to a problem? Please describe.**
In maps with progress triggers, currently the only way to go back to the last progress trigger is by touching a `trigger_momentum_teleport_progress`

**Describe the solution you'd like**
Add a command that replicates the behavior of `trigger_momentum_teleport_progress`

### PR DESCRIPTION
**Pull request by sleepCOW**
Closes #897 

Added new console command
name: mom_prog_tele
effect: teleports the player to the last touched progress trigger
note: command will set your velocity to zero and would keep the player's rotation

PR to the docs: https://github.com/momentum-mod/docs/pull/67

### 

- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

File: mp/src/game/server/momentum/mom_triggers.cpp [hunk 1]
```diff
@@ -22,6 +22,28 @@ static MAKE_TOGGLE_CONVAR(mom_triggers_overlay_bbox_enable, "0", FCVAR_DEVELOPME
 static MAKE_TOGGLE_CONVAR(mom_triggers_overlay_text_enable, "0", FCVAR_DEVELOPMENTONLY,
                           "Toggles showing the entity text for momentum triggers, needs map restart if changed!\n");
 
+CON_COMMAND(mom_trigger_teleport_progress, "Teleports player to the last progress trigger\n")
```
Comment (braem) [inline mp/src/game/server/momentum/mom_triggers.cpp:25]: To keep the naming of progress related commands I suggest renaming this to something like `mom_prog_tele`, and moving it into [`mom_system_progress.cpp`](https://github.com/momentum-mod/game/blob/e3efc48eb58cf56af61b2fef8d0bf7e3bf754680/mp/src/game/server/momentum/mom_system_progress.cpp).

File: mp/src/game/server/momentum/mom_triggers.cpp [hunk 2]
```diff
@@ -22,6 +22,28 @@ static MAKE_TOGGLE_CONVAR(mom_triggers_overlay_bbox_enable, "0", FCVAR_DEVELOPME
...
+{
+    CMomentumPlayer *pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (pPlayer)
```
Comment (braem) [inline mp/src/game/server/momentum/mom_triggers.cpp:28]: Can early return this here and remove the "Player is null" DevWarning, eg.
```
if (!pPlayer)
    return;
```

File: mp/src/game/server/momentum/mom_triggers.cpp (hunk 1 above)
Comment (sleepCOW) [reply mp/src/game/server/momentum/mom_triggers.cpp]: A good suggestion I have renamed it and moved to mom_system_progress.cpp

File: mp/src/game/server/momentum/mom_triggers.cpp (hunk 2 above)
Comment (sleepCOW) [reply mp/src/game/server/momentum/mom_triggers.cpp]: Did this if-statement for better branch prediction but because this code isn't bottle-neck, followed your advice for better readability, thanks a lot!

File: mp/src/game/server/momentum/mom_triggers.cpp [hunk 3]
```diff
@@ -22,6 +22,7 @@ static MAKE_TOGGLE_CONVAR(mom_triggers_overlay_bbox_enable, "0", FCVAR_DEVELOPME
 static MAKE_TOGGLE_CONVAR(mom_triggers_overlay_text_enable, "0", FCVAR_DEVELOPMENTONLY,
                           "Toggles showing the entity text for momentum triggers, needs map restart if changed!\n");
 
+
```
Comment (Gocnak) [inline mp/src/game/server/momentum/mom_triggers.cpp:25]: Unnecessary change

File: mp/src/game/server/momentum/mom_system_progress.cpp [hunk 4]
```diff
@@ -86,5 +88,25 @@ CON_COMMAND(mom_prog_reset, "Resets progress")
     g_pMomentumProgress->Reset();
 }
 
+CON_COMMAND(mom_prog_tele, "Teleports player to the last progress trigger\n")
```
Comment (Gocnak) [inline mp/src/game/server/momentum/mom_system_progress.cpp:91]: This should be inside the mom_player class, as this  class is actually an older one used for campaign progress. This command is a utility function for the player, so it makes sense in the player or timer class.

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: mp/src/game/server/momentum/mom_triggers.cpp (hunk 3 above)
Comment (sleepCOW) [reply mp/src/game/server/momentum/mom_triggers.cpp]: Sorry for this one, removed it

File: mp/src/game/server/momentum/mom_system_progress.cpp (hunk 4 above)
Comment (sleepCOW) [reply mp/src/game/server/momentum/mom_system_progress.cpp]: Moved this command to mom_player.cpp but kept functionality inside. Does an additional member function inside of player class is required for such a small thing?

Comment (Gocnak) [reply mp/src/game/server/momentum/mom_system_progress.cpp]: Nah it's fine being in the CON_COMMAND 👍

**Review by Gocnak — APPROVED** [review]
